src/main/java/net/minestom/server/inventory/Inventory.java

```diff
@@ -58,11 +58,12 @@ public Inventory(@NotNull InventoryType inventoryType, @NotNull String title) {
     }
 
     private static byte generateId() {
-        final byte id = (byte) Math.abs((byte) ID_COUNTER.incrementAndGet());
-        if (id == 0) { // zero is player's inventory id
-            return generateId();
+        int id = ID_COUNTER.incrementAndGet();
+        if (id > 127) {
+            ID_COUNTER.set(1);
+            id = 1;
```

Comment:
Start `ID_COUNTER` at 0, then get the ID using `final byte id = (byte)ID_COUNTER.updateAndGet(i -> i == 128 ? 1 : i + 1);`

Setting the value to 1 then doing `id = 1` could potentially be unsafe, if multiple threads are generating an ID at the same time you could get multiple inventories with id = 1.